common/app/templates/inlineJS/blocking/shouldEnhance.scala.js

```diff
@@ -1,54 +1,98 @@
 @(item: model.MetaData)(implicit request: RequestHeader)
 @import conf.switches.Switches._
 
-(function (navigator, window) {
-    // Enable manual optin to core functionality/optout of enhancement
-    var personPrefersCore = function () {
-        var locationHash = window.location.hash;
-        if (locationHash === '#featuresoff' || locationHash === '#core' || locationHash === '#gu.prefs.force-core=on') return true;
-        if (locationHash === '#featureson' || locationHash === '#nocore' || locationHash === '#gu.prefs.force-core=off') return false;
-        try {
-            var preference = window.localStorage.getItem('gu.prefs.force-core') || 'off';
-            return /"value":"on"/.test(preference);
-        } catch (e) {
-            return false;
-        }
+// Determine whether we want to run the enhanced app or not.
+// It can come either from a preference in localStorage (see `enhancedKey`)
+// or from a temporary opt-in or -out override, using `#standard` or `#enhanced` hash fragments.
+
+(function (window) {
+    var location = window.location;
+    var hash = location.hash;
+
+    var navigator = window.navigator;
+    var platform = navigator.platform;
+
+    var isFront = @item.isFront;
+    var enhancedKey = 'gu.prefs.enhanced';
+
+    var preferEnhanced;
+    try {
+        preferEnhanced = JSON.parse(localStorage.getItem(enhancedKey));
+    } catch (e) {
+        preferEnhanced = null;
     };
 
-    var personDemandsFeatures = function () {
-        var locationHash = window.location.hash;
-        if (locationHash === '#featureson' || locationHash === '#nocore' || locationHash === '#gu.prefs.force-core=off') return true;
+    // This has been called core, featuresoff, universal etc, and a few previous ways of
+    // opting in and out existed.
+    // Now we have settled on standard and enhanced, the previous methods are handled
+    // here, temporarily. This function can go after a while.
+    function normaliseNomenclature() {
+        var coreKey = 'gu.prefs.force-core', settingCorePref, corePref;
+
+        // update any `force-core` stored pref
         try {
-            var preference = window.localStorage.getItem('gu.prefs.force-core') || 'unknown';
-            return /"value":"off"/.test(preference);
-        } catch (e) {
-            return false;
+            var localStorage = window.localStorage;
+            if (corePref = localStorage.getItem(coreKey)) {
```

Comment:
Nitpicking here, but I think it hurts readability to combine an assignment with an if statement, just because it's more complex.